Packet++/header/GvcpLayer.h

```diff
@@ -0,0 +1,772 @@
+
+		// Asynchronous Events
+		EventCmd = 0x00C0,
+		EventAck = 0x00C1,
+		EventDataCmd = 0x00C2,
+		EventDataAck = 0x00C3,
+
+		// Miscellaneous
+		ActionCmd = 0x0100,
+		ActionAck = 0x0101,
+		Unknown = 0xFFFF
+	};
+
+	std::ostream& operator<<(std::ostream& os, GvcpCommand command);
+
+	/// @brief Gvcp response status
+	/// See spec "Table 19-1: List of Standard Status Codes"
+	enum class GvcpResponseStatus : uint16_t
+	{
+		Success = 0x0000,
+		PacketResend = 0x0100,
+		NotImplemented = 0x8001,
+		InvalidParameter = 0x8002,
+		InvalidAddress = 0x8003,
+		WriteProtect = 0x8004,
+		BadAlignment = 0x8005,
+		AccessDenied = 0x8006,
+		Busy = 0x8007,
+		LocalProblem = 0x8008,     // deprecated
+		MsgMismatch = 0x8009,      // deprecated
+		InvalidProtocol = 0x800A,  // deprecated
+		NoMsg = 0x800B,            // deprecated
+		PacketUnavailable = 0x800C,
+		DataOverrun = 0x800D,
+		InvalidHeader = 0x800E,
+		WrongConfig = 0x800F,  // deprecated
+		PacketNotYetAvailable = 0x8010,
+		PacketAndPrevRemovedFromMemory = 0x8011,
+		PacketRemovedFromMemory = 0x8012,
+		NoRefTime = 0x8013,                     // GEV 2.0
+		PacketTemporarilyUnavailable = 0x8014,  // GEV 2.0
+		Overflow = 0x8015,                      // GEV 2.0
+		ActionLate = 0x8016,                    // GEV 2.0
+		LeaderTrailerOverflow = 0x8017,         // GEV 2.1
+		Error = 0x8FFF,
+		Unknown = 0xFFFF
+	};
+
+	std::ostream& operator<<(std::ostream& os, GvcpResponseStatus status);
+
+#pragma pack(push, 1)
+	/// @brief Gvcp request header
+	/// @note refer to the spec "15.1 Request Header". The data is stored as big-endian.
+	struct GvcpRequestHeader
+	{
+		friend class GvcpRequestLayer;
+
+	protected:
+		uint8_t magicNumber = internal::kGvcpMagicNumber;  // always fixed
+		uint8_t flag = 0;  // 0-3 bits are specified by each command, 4-6 bits are reserved, 7 bit is acknowledge
+		uint16_t command = 0;
+		uint16_t dataSize = 0;
+		uint16_t requestId = 0;
+
+	public:
+		// ------------- methods --------------
+		GvcpRequestHeader() = default;
+
+		GvcpRequestHeader(GvcpFlag flag, GvcpCommand command, uint16_t dataSize, uint16_t requestId)
+		    : flag(flag), command(hostToNet16(static_cast<uint16_t>(command))), dataSize(hostToNet16(dataSize)),
+		      requestId(hostToNet16(requestId))
+		{}
+
+		GvcpCommand getCommand() const
+		{
+			return static_cast<GvcpCommand>(netToHost16(command));
+		}
+	};
+	static_assert(sizeof(GvcpRequestHeader) == internal::kGvcpRequestHeaderLength,
+	              "Gvcp request header size should be 8 bytes");
+
+	/// @brief Gvcp discovery request
+	struct GvcpDiscoveryRequest : public GvcpRequestHeader
+	{
+		// no addition fields
+
+		// ------------- methods --------------
+
+		/**
+		 * @brief Check if the broadcast is allowed
+		 * @return true The broadcast acknowledge is allowed
+		 */
+		bool hasAllowBroadcastFlag() const
+		{
+			constexpr GvcpFlag kAllowBroadcastFlag = 0b0001000;
+			return (flag & kAllowBroadcastFlag) == kAllowBroadcastFlag;
+		}
```

Comment:
Why do we need this method in the struct?

Reply:
same